projects/server-database/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"server-database/types"
+	"strconv"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/joho/godotenv"
+)
+
+func handleImages(w http.ResponseWriter, r *http.Request) {
+	conn, err := setDatabaseConnection()
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+	}
+	defer conn.Close(context.Background())
+	images, err := fetchImages(conn)
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+	}
+	indent := getIndentParam(r)
+	b, err := json.MarshalIndent(images, "", indent)
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(b)
+}
+
+func getIndentParam(r *http.Request) string {
+	params := r.URL.Query()
+
+	indentSize, err := strconv.Atoi(params.Get("indent"))
+	if err != nil {
+		fmt.Fprint(os.Stderr, err)
+	}
+	indent := ""
+	for i := 0; i < indentSize; i++ {
+		indent += " "
+	}
+	return indent
+}
+
+func fetchImages(conn *pgx.Conn) ([]types.Image, error) {
+	var images []types.Image
+	rows, err := conn.Query(context.Background(), "SELECT title, url, alt_text FROM public.images")
```

Comment:
In a real production workload we would need some sensible way to limit the number of responses, such as requirng query params, using pagination techniques, etc. Here you might add a LIMIT clause to your query.

You don't need to go off into the weeds with pagination here - it is out of scope for the exercise - but just in case you're curious, here is some more info on what I'm talking about
https://medium.easyread.co/how-to-do-pagination-in-postgres-with-golang-in-4-common-ways-12365b9fb528